### PR TITLE
Engine upgrade

### LINKED ISSUE #44: Here comes the crash
At [8d630b3](https://github.com/MustaphaTR/Generals-Alpha/commit/8d630b307c42fc0c34fe3b9ca811bc4c1fc0f43a)

A crash related to SharedCargo, using deploy order.

```
OpenRA engine version 286e100caa3de0374682717625982ec0c21fdeec
OpenRA Language: en
Generals Alpha mod version gen-20230705
on map b3cb982f715e98b4183a7ce7505fee21a506d44a (Burning Plain by james.bong).
Date: 2023-09-07 13:58:53Z
Operating System: Windows (X64, Microsoft Windows NT 10.0.19045.0)
Runtime Version: .NET CLR 6.0.11
Installed Language: zh (Installed) zh (Current) zh (Current UI)
Exception of type `System.ArgumentException`: An item with the same key has already been added. Key: vehicle.quad_cannon 3736
   at System.Collections.Generic.Dictionary`2.TryInsert(TKey key, TValue value, InsertionBehavior behavior)
   at OpenRA.Mods.Common.Traits.GivesBounty.OpenRA.Mods.Common.Traits.INotifyPassengerEntered.OnPassengerEntered(Actor self, Actor passenger) in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Mods.Common\Traits\GivesBounty.cs:line 87
   at OpenRA.Mods.AS.Traits.SharedCargo.Load(Actor self, Actor a) in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Mods.AS\Traits\SharedCargo.cs:line 330
   at OpenRA.Mods.AS.Activities.RideSharedTransport.<>c__DisplayClass6_0.<OnEnterComplete>b__0(World w) in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Mods.AS\Activities\RideSharedTransport.cs:line 69
   at OpenRA.World.Tick() in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Game\World.cs:line 453
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Game\Game.cs:line 631
   at OpenRA.Game.LogicTick() in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Game\Game.cs:line 646
   at OpenRA.Game.Loop() in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Game\Game.cs:line 815
   at OpenRA.Game.Run() in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Game\Game.cs:line 868
   at OpenRA.Game.InitializeAndRun(String[] args) in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Game\Game.cs:line 305
   at OpenRA.Launcher.Program.Main(String[] args) in D:\work\Projects\Github\Generals-Alpha\engine\OpenRA.Launcher\Program.cs:line 32
```
[replay.zip](https://github.com/MustaphaTR/Generals-Alpha/files/12551288/replay.zip)

Update: 

In short:

`vehicle.quad_cannon 3736`, produced at 57184 ticks, at cyan color player, Stealth General.

building.tunnel_network 3810->SharedCargo: It is Loaded at: 58394, it is in world? True
building.tunnel_network 3810->GivesBounty: It is loaded in: 58394
building.tunnel_network.no_free_actor 3842->SharedCargo: INotifyEnteredSharedCargo initialized: 58431, it is in world? False
building.tunnel_network.no_free_actor 3842->GivesBounty: It is loaded in: 58431
building.tunnel_network 3810->SharedCargo: It is Unloaded at: 60595, it is in world? False
building.tunnel_network 3810->GivesBounty: It is unloaded at: 60595
building.tunnel_network.no_free_actor 3842->SharedCargo: It is Loaded at: 61005, it is in world? True
building.tunnel_network.no_free_actor 3842->GivesBounty: It is loaded in: 61005

### PR DESCRIPTION
**Pull request by dnqbob**
fix #44

### CODE REVIEW

File: mods/gen/rules/vehicles.yaml [hunk 1]
```diff
@@ -224,11 +224,6 @@ vehicle.humvee:
 		Weapons: debris.humvee.turret
 		Pieces: 1, 1
 		Range: 1c0, 1c0
-	ThrowsShrapnel@Turret2:
-		Weapons: debris.rocket_buggy.turret
-		Pieces: 1, 1
-		Range: 1c0, 1c0
-		RequiresCondition: tow_missile
```
Comment (MustaphaTR) [inline mods/gen/rules/vehicles.yaml:231]: I can't check the new artwork rn, but it should still throw the Turret on death imo. After all even the small MG Turret is throw rn. Just needs a new weapon.

File: mods/gen/rules/player/ai.yaml [hunk 2]
```diff
@@ -535,158 +535,165 @@ Player:
 		PassengerTypes: infantry.hacker, infantry.super_hacker
 
 	SupportPowerBotASModule:
-		RequiresCondition: enable-any-ai
+		RequiresCondition: enable-normal-ai || enable-hard-ai
```
Comment (MustaphaTR) [inline mods/gen/rules/player/ai.yaml:542]: Not sure if i like all SWs being disabled to Easy AI like this. Especially for PC/Scud/Nuke, would be weird that AI still builds them but not fire. Making them not build SWs/Not take powerful GPs would be better but we merged the Production Modules.

File: mods/gen/rules/vehicles.yaml (hunk 1 above)
Comment (dnqbob) [reply mods/gen/rules/vehicles.yaml:231]: Just add a white line on the turret, so throwing that will be weird

File: mods/gen/rules/player/ai.yaml (hunk 2 above)
Comment (dnqbob) [reply mods/gen/rules/player/ai.yaml:542]: If my memory is correct, in General easy AI won't use support power. 

In addition, it is kinda important for prc to build its super weapon as there are 3 important upgrades need to be brought here, and if easy prc can have super weapon, we should also allow easy USA and gla to have their super weapon.

Comment (dnqbob) [reply mods/gen/rules/player/ai.yaml:542]: I was thinking, we should still make main logic of general ranking system code in c# and expose script interface to show it on UI, in doing so we can write bot module to let the bot choose properly between General Skill and thus have different strategy.

**Comment by MustaphaTR** [comment]
Need rebase, it seems.

File: mods/gen/rules/player/ai.yaml (hunk 2 above)
Comment (MustaphaTR) [reply mods/gen/rules/player/ai.yaml:542]: I didn't think of PRC upgrades. But i'm still not a big fan of making the Easy AI build them but not fire, even if they didn't in the original.

**Comment by dnqbob** [comment]
Rebased

**Pull request merged**